akka-stream-tests/src/test/scala/akka/stream/scaladsl/GraphPartitionSpec.scala

```diff
@@ -177,4 +177,17 @@ class GraphPartitionSpec extends StreamSpec {
     }
 
   }
+
+  "divertTo must send matching elements to the sink" in assertAllStagesStopped {
+    val odd = TestSubscriber.probe[Int]()
+    val even = TestSubscriber.probe[Int]()
+    Source(1 to 2).divertTo(Sink.fromSubscriber(odd), _ % 2 != 0).to(Sink.fromSubscriber(even)).run()
+    even.request(1)
+    even.expectNoMsg(1.second)
+    odd.request(1)
+    odd.expectNext(1)
+    even.expectNext(2)
+    odd.expectComplete()
+    even.expectComplete()
+  }
```

Comment:
Would be good with test coverage for what is expected when the respective downstreams cancel also. (even if that comes implicitly from what partition does)

Reply:
Since it piggybacks on Partition, I chose to do the same as alsoTo does, for consistency.

Reply:
I'd still like a test case verifying that, to cover regression etc. Note that we have a recent open issue about how `alsoTo` behaves where I think we agreed that eager cancelation would be less surprising: #24291

Reply:
Yes, I'd also find it weird  that partition only cancels if *all* downstreams cancel. I mean, what will happen to elements destined for cancelled sinks?

Reply:
Looking at the stage, it seems elements for closed downstreams goes to /dev/null